int64 x
int64 y
int64 z
int64 status
int64 face
